### PR TITLE
[tinyfsm] Add new port

### PR DESCRIPTION
**Pull request by koprok**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/tinyfsm/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-tinyfsm)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
```
Comment (dg0yt) [inline ports/tinyfsm/portfile.cmake:18]: IIUC this port is header-only. So just completely skip the debug variant by adding (before `vcpkg_cmake_configure`):
~~~cmake
set(VCPKG_BUILD_TYPE release) # header-only
~~~

**Comment by dg0yt** [comment]
Actually I wouldn't even create a CMake project or config. The header can be simply used via `find_path`, `target_include_directories`. This can be documented in a `usage` file, and it wouldn't lock users into vcpkg.

File: ports/tinyfsm/portfile.cmake (hunk 1 above)
Comment (koprok) [reply ports/tinyfsm/portfile.cmake]: Done as you suggested.

**Comment by koprok** [comment]
> Actually I wouldn't even create a CMake project or config. The header can be simply used via `find_path`, `target_include_directories`. This can be documented in a `usage` file, and it wouldn't lock users into vcpkg.

@dg0yt , I find the exported CMake config useful:
- to make libraries usage consistent, no matter if they are header-only
- I found that Vcpkg installs a CMake config for other simple header-only libraries (e.g. sajson, nonius, concurrentqueue) and I took a similar approach.

Also, installing a CMake config does not stop the user from using `find_path` manually if they prefer so.

Anyway, I will remove the CMake config if you prefer so, should I?

**Comment by dg0yt** [comment]
> Anyway, I will remove the CMake config if you prefer so, should I?

I don't have any official word here. But I'm against promoting *unofficial* patterns of usage for no practical benefit. This a narrow interpretation of the following guideline:
*"A core design ideal of vcpkg is to not create "lock-in" for customers."*
https://learn.microsoft.com/en-us/vcpkg/contributing/maintainer-guide#add-cmake-exports-in-an-unofficial--namespace

(Some header-only packages have official (upstream) CMake config. And some real libraries are hard to use without CMake config. These are fair use cases.)

**Comment by koprok** [comment]
> I don't have any official word here. But I'm against promoting _unofficial_ patterns of usage for no practical benefit.

Ok, @dg0yt, I have removed the exported CMake config. I hope this will get the port merged.
The automatically-generated usage instructions are good enough, so there is no need for `usage` file.

**Review by JonLiu1993 — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**